From b8387f2d170111c377fe024164719e976d9d2a36 Mon Sep 17 00:00:00 2001
From: Tapasweni Pathak <tapaswenipathak@gmail.com>
Date: Wed, 6 Mar 2019 23:27:12 +0530
Subject: [PATCH] drivers: dma-bug: Add null dereference check

seqno_fence should have a null check before being dereferenced as
to_seqno_fence can return null.

---
 drivers/dma-buf/seqno-fence.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/dma-buf/seqno-fence.c b/drivers/dma-buf/seqno-fence.c
index a4e783a..2ed33e7 100644
--- a/drivers/dma-buf/seqno-fence.c
+++ b/drivers/dma-buf/seqno-fence.c
@@ -31,14 +31,16 @@ static const char *seqno_fence_get_driver_name(struct dma_fence *fence)
 static const char *seqno_fence_get_timeline_name(struct dma_fence *fence)
 {
 	struct seqno_fence *seqno_fence = to_seqno_fence(fence);
-
+  if (!seqno_fence)
+    return false;
 	return seqno_fence->ops->get_timeline_name(fence);
 }
 
 static bool seqno_enable_signaling(struct dma_fence *fence)
 {
 	struct seqno_fence *seqno_fence = to_seqno_fence(fence);
-
+  if (!seqno_fence)
+    return false;
 	return seqno_fence->ops->enable_signaling(fence);
 }
 
@@ -46,6 +48,8 @@ static bool seqno_signaled(struct dma_fence *fence)
 {
 	struct seqno_fence *seqno_fence = to_seqno_fence(fence);
 
+	if (!seqno_fence)
+    return false;
 	return seqno_fence->ops->signaled && seqno_fence->ops->signaled(fence);
 }
 
-- 
2.10.0

